neighbors/src/components/AddPost/AddPost.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { addDoc } from 'firebase/firestore';
+
+import { ThemeContext, needHelpPostsData, offerHelpPostsData } from '../../helpers/apiCommunication';
+export const AddPost = () => {
+  const { user } = useContext(ThemeContext); // Zalogowany user na potrzeby testów
+  const [chooseTypeOfPost, setTypeOfPost] = useState('');
+  const [notAvailbeToWritePost, setAvailbeToWritePost] = useState(true);
+  const [postData, setPostData] = useState({
+    post: '',
+    postTitle: '',
+  });
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    setPostData({
+      ...postData,
+      userFirstName: user.firstName,
+      userLastName: user.lastName,
+      userPhoneNumber: user.phoneNumber,
+      userCity: user.city,
+      userStreet: user.street,
+      userEmail: user.email,
+      [name]: value,
+    });
+  };
+  const onChangleSelectPost = (e) => {
+    setTypeOfPost(e.target.value);
+  };
+  useEffect(() => {
+    if (chooseTypeOfPost === 'NeedHelp' || chooseTypeOfPost === 'OfferHelp') {
+      setAvailbeToWritePost(false);
+    } else {
+      setAvailbeToWritePost(true);
+    }
+  }, [chooseTypeOfPost]);
+
+  const onClick = (e) => {
+    e.preventDefault();
+    if (chooseTypeOfPost === 'NeedHelp') {
+      addDoc(needHelpPostsData, postData).then(() => {});
+    }
+    if (chooseTypeOfPost === 'OfferHelp') {
+      addDoc(offerHelpPostsData, postData).then(() => {});
+    }
+    setPostData({ post: '', postTitle: '' });
+  };
+  return (
+    <>
+      <form>
+        <label htmlFor="TypeOfPost">Choose type of post: </label>
+        <select name="typOfPost" onChange={onChangleSelectPost}>
+          <option value=" "></option>
+          <option value="NeedHelp">Need help</option>
+          <option value="OfferHelp">Offer help</option>
+        </select>
+        <label htmlFor="postTitle">Add title: </label>
+        <input
+          type="text"
+          disabled={notAvailbeToWritePost}
+          value={postData.postTitle}
+          name="postTitle"
+          onChange={onChange}
+        ></input>
```

Comment:
Jeżeli komponent nie ma children to powinnen być self-closing (to się pojawia w wielu miejscach, dla inputa, option, textArea)

Reply:
Poprawione zarówno  w AddPost jak i w  komponencie rejestracji.